test/docker-base.bats

```diff
@@ -31,3 +31,13 @@ EOF
     umoci unpack --image oci:layer1 dest
     [ ! -f dest/rootfs/favicon.ico ]
 }
+
+@test "image with empty layer" {
+    cat > stacker.yaml <<EOF
+image:
+    from:
+        type: docker
+        url: docker://grafana/grafana-oss:10.1.2-ubuntu
```

Comment:
How do we know this image will stay around and will also have empty layers?  Would it be better to create an an image with an empty layer in the test-suite itself?

Reply:
Good question, we can push it to stacker ghcr to preserve it. Some strange combination of tooling - it is dockerv2 schema which still contains schemav1-type layers.

Reply:
https://github.com/project-stacker/stacker/pull/525

Would rather take an actual example.

Reply:
And not quite the same. The layer exists, but when you untar, it has zero changes to the rootfs.

Reply:
OK, seems like we can create it somehow (docker certainly does).  

Right, so I copied out the empty tar.gz file from that graphana image...

Let me get that to you to test.

Reply:
I think this should work, but you said it didn't.  Does stacker not detect the empty layer in here?

```
#!/bin/bash
OCID=oci_dir
umoci init --layout "$OCID"
umoci new --image "$OCID:emptylayer"

EMPTY_TGZ="empty-layer.tar.gz"
base64 --decode > "$EMPTY_TGZ" << EOF
H4sIAAAAAAAA/2IYBaNgFIxYAAgAAP//Lq+17wAEAAA=
EOF

umoci raw add-layer --image "$OCID:emptylayer" \
    --history.author docker \
    --history.comment "empty tar" \
    "$EMPTY_TGZ"
    
umoci gc --layout "$OCID"
```

maybe the GC helps?

Reply:
This seems to be creating an image with a single empty layer. Don't think there is any "legal" image that has that.